flocker/control/_model.py

```diff
@@ -110,23 +111,33 @@ def pmap_field(
     :param initial: An initial value for the field.  This will first be coerced
         using the field's factory.  If not given, the initial value is an empty
         map.
+    :param factory: A factory used to convert input arguments to the stored
+        value whenever it is set. Note that this will be composed with the
+        constructor for the ``CheckedPMap`` class constructed for this field.
```

Comment:
Ah, no. I get it. The factory is used when assigning a new value for the pmap_field attribute, not an element in the pmap.